### ISSUE TITLE
Add a button on Cocoon/build dashboard to manually close the tree.

### ISSUE THREAD

**Issue by christopherfujino** [flutter/flutter#74529]
The Framework Gardener should be able to manually close the tree, even if tests are passing. In the case of https://github.com/flutter/flutter/issues/74431, a commit was landed to explicitly fail a test to achieve this purpose.

Ideally, there would be a UI that would allow an authenticated user to manually close the tree, linking a GitHub issue.

**Comment by CaseyHillers** [comment]
@dnfield started a PR for the backend call: https://github.com/flutter/cocoon/pulls

**Comment by dnfield** [comment]
I need to get back to that at some point, but if anyone wants to take it over I'd be happy to sync up on where it was left.

**Comment by Hixie** [comment]
Given that people are checking in broken tests to work around this, this is a much higher priority.

**Comment by Hixie** [comment]
cc @godofredoc

**Comment by godofredoc** [comment]
This is blocked on enforcing the auto-submit bot to land all the PRs

**Comment by godofredoc** [comment]
Enforcing the auto-submit bot is scheduled for Q42021

**Comment by CaseyHillers** [comment]
I'll try this and see where we're at. From a glance it looks like:

1. Add a TreeStatus datastore entity
  A. Update it on the push build status handlers
  B. Return it on the get-build-status handler
2. Nice to have: Special log whenever the tree state changes

**Comment by godofredoc** [comment]
I'd suggest to keep it as simple as we can. E.g having a model completely independent of the actual tree status that the autosubmit bot can check to see if a given tree is manually open/closed and then submit PRs if it open or skip if it is closed.

**Comment by CaseyHillers** [comment]
Moving out of ticket queue as I won't get to this during the code freeze.

After https://github.com/flutter/cocoon/pull/1527, it'll be reasonable to add a generic `TreeStatus` entity that can be used to override the tree status.

**Comment by godofredoc** [comment]
This is a feature request, updating as P3

**Comment by CaseyHillers** [comment]
@XilaiZhang is going to work on this next quarter to help give the rollers more control

**Comment by godofredoc** [comment]
@XilaiZhang should this be still assigned to you?

**Comment by ricardoamador** [comment]
@keyonghan we should talk about reprioritizing this at our triage meeting tomorrow. With the increase in Github webhook degradation events it makes sense to protect ourselves from allowing changes in without running all the check runs.
